### PR TITLE
New CLI for creating Addon Boilerplate #118

### PR DESCRIPTION
**Pull request by Znackt**
Added a new feature to help developers minimize their time and effort writing boilerplates for creating their projects! 
Fixes the issue #118 

I chose to write program using this way because, to help the users not waste their time on writing boilerplate code, 
I have solved each feature as suggested, 
The new command `meta-system create-addon` has been implemented.
I have not implemented the wizard.
The output of the code works in current directory.

### CODE REVIEW

File: template/addons/entrypoint.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Imported example functions
+import { createTokenFunction, hashFunction, matchesHashFunction, verifyTokenFunction } from "./definitions.js";
```
Comment (zelcion) [inline template/addons/entrypoint.js:2]: These functions are imported but their files are not declared. I see you copied them from the Meta-Hash Addon, which is great, but remember to cleanup and adapt the code to your goals after copying.

File: template/addons/entrypoint.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    // Register functions with the broker
+    const functions = [createTokenFunction, verifyTokenFunction, hashFunction, matchesHashFunction];
+
+    for (const func of functions) {
+        broker.addonsFunctions.register(func.function, func.definition);
+    }
+
+    // Perform additional setup based on configuration
+    if (configuration.parameter1) {
+        console.log(`Configuration parameter1: ${configuration.parameter1}`);
```
Comment (zelcion) [inline template/addons/entrypoint.js:14]: We can instead use the broker logger here as `broker.logger.info()`

File: template/addons/entrypoint.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    // Signal that setup is complete
+    broker.done();
+};
+
+export const boot = (broker, configuration) => {
+    // Initialize addon functionality
+    console.log("Addon is booting up with configuration:", configuration);
+
+    // Example of starting a server or other continuous behavior
+    // server.listen(configuration.port);
```
Comment (zelcion) [inline template/addons/entrypoint.js:30]: This kind of comment is really useful, so I'd give it a bit more context.

What about we increment it like: "The boot function is used to start continuous behavior, like listening to HTTP calls,  polling, and more"

File: template/addons/package.json
Comment (zelcion) [inline template/addons/package.json:1]: I think we can drop this "package.json" :wink: 
This is not required for the boilerplate to work.

File: tsconfig.json [hunk 4]
```diff
@@ -8,5 +8,5 @@
   "include": [
     "src/**/*",
     "test/**/*"
-  ]
+, "template"  ]
```
Comment (zelcion) [inline tsconfig.json:11]: including this in the tsconfig means the code will be transpiled. Was this the plan here?

**Review by zelcion — CHANGES_REQUESTED** [review]
I really appreciate the contribution! Take a look at the suggestions and questions, so we can move on with the fix! :muscle:

The more important aspect of this PR is about adding new libraries, which we should avoid unless really required.

Also, tests are really welcome! Ping me if you need help creating a new test!

**Comment by Znackt** [comment]
Sure! I'm working on this :)

**Pull request closed without merging**